File: 121-Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        _min = float('inf')
        res = 0
        for i in range(len(prices)):
            _min = min(_min, prices[i])
            res = max(res, prices[i]-_min)
            
        return res

File: JavaScript/121-Best Time to Buy and Sell Stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let res = 0, min = Number.POSITIVE_INFINITY
    for (let price of prices){
        min = Math.min(min, price);
        res = Math.max(price-min, res);
    }
    return res;
};